lib/CModHandler.cpp

```diff
@@ -532,6 +532,51 @@ JsonNode addMeta(JsonNode config, std::string meta)
 	return config;
 }
 
+CModInfo::Version CModInfo::Version::GameVersion()
+{
+	return Version(GameConstants::VCMI_VERSION_MAJOR, GameConstants::VCMI_VERSION_MINOR, GameConstants::VCMI_VERSION_PATCH);
+}
+
+CModInfo::Version CModInfo::Version::fromString(std::string from)
+{
+	int major = 0, minor = 0, patch = 0;
+	try
+	{
+		auto pointPos = from.find('.');
+		major = std::stoi(from.substr(0, pointPos));
+		if(pointPos != std::string::npos)
+		{
+			from = from.substr(pointPos + 1);
+			pointPos = from.find('.');
+			minor = std::stoi(from.substr(0, pointPos));
+			if(pointPos != std::string::npos)
+				patch = std::stoi(from.substr(pointPos + 1));
+		}
+	}
+	catch(const std::invalid_argument & e)
+	{
+		return Version();
+	}
+	return Version(major, minor, patch);
+}
+
+std::string CModInfo::Version::toString() const
+{
+	return std::to_string(major) + '.' + std::to_string(minor) + '.' + std::to_string(patch);
+}
+
+bool CModInfo::Version::compatible(const Version & other, bool checkMinor, bool checkPatch) const
+{
+	return  (major == other.major &&
+			(!checkMinor || minor >= other.minor) &&
+			(!checkPatch || minor > other.minor || (minor == other.minor && patch >= other.patch)));

```

Comment:
```suggestion
	return (major == other.major
		&& (!checkMinor || minor >= other.minor)
		&& (!checkPatch || minor > other.minor || (minor == other.minor && patch >= other.patch)));

```

Reply:
I prefer style with operators in the end of line, because It's easy to see that line is not ended.
As for indends, GitHub uses 8 spaces as one tab, instead of 4

Reply:
operators are placed in the beginning of a line mainly to make it easier to see which operator is used, which is especially convenient when lines are long. Not insisting on changing.

> As for indends, GitHub uses 8 spaces as one tab, instead of 4

use mouse to see that lines after `return` are indented by 2 tabs instead of 1 ;)

Reply:
> beginning of a line mainly to make it easier to see which operator is used

For me it's opposite:)

Reply:
Regarding indentation our code style clearly insist on using tabs not spaces

Reply:
there are tabs